sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveSessionStateBuilder.scala

```diff
@@ -98,20 +98,7 @@ class HiveSessionStateBuilder(session: SparkSession, parentState: Option[Session
       override def extraPlanningStrategies: Seq[Strategy] =
         super.extraPlanningStrategies ++ customPlanningStrategies
 
-      override def strategies: Seq[Strategy] = {
-        experimentalMethods.extraStrategies ++
-          extraPlanningStrategies ++ Seq(
-          FileSourceStrategy,
-          DataSourceStrategy(conf),
-          SpecialLimits,
-          InMemoryScans,
-          HiveTableScans,
-          Scripts,
-          Aggregation,
-          JoinSelection,
-          BasicOperators
-        )
-      }
+      override def strategies: Seq[Strategy] = Seq(HiveTableScans, Scripts) ++ super.strategies
```

Comment:
This breaks the assumption that `experimentalMethods.extraStrategies` should always run first.

I think we can just do:
```
override def extraPlanningStrategies: Seq[Strategy] =
  super.extraPlanningStrategies ++ customPlanningStrategies ++ Seq(HiveTableScans, Scripts)
```

Reply:
OK, let me update it.